python/BioSimSpace/Process/_somd.py

```diff
@@ -577,7 +578,8 @@ def _generate_config(self):
             self.addToConfig("lambda_val = %s" \
                 % self._protocol.getLambda())                                       # The value of lambda.
 
-            res_num = self._system.search("resname LIG")[0]._sire_object.number().value()
+            res_num = _System(self._renumbered_system) \
+                .search("resname LIG")[0]._sire_object.number().value()
```

Comment:
Is it better to detect it by searching for perturbable residues instead and asserting that there is only one? Relying on resname is probably a bit unreliable.

Reply:
For SOMD, it requires a specific name, i.e. `LIG` in this case, which is enforced elsewhere during the merging. I agree that it's probably not super reliable depending on how the system was generated, though. I'll change the search to make it more robust.

Cheers.

Reply:
Oh, and we also enforce that there only a single perturbable molecule here:

https://github.com/michellab/BioSimSpace/blob/950c35a95d08ad0d3e7c6fd310e4d49dd61def9e/python/BioSimSpace/Process/_somd.py#L225-L226

This is before the `res_num` is computed.